### PR TITLE
Move Welcome Notification to Active Jobs

### PR DESCRIPTION
**Pull request by cyrillefr**
- creation of welcome notification is in own service
 - which is called by an active job

Resolves: #1996

## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Documentation Update

## Description
Move Welcome Notification to Active Jobs
## Related Tickets & Documents
Issue #1996 
## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: spec/jobs/notifications/welcome_notification_job_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+require "rails_helper"
+
+RSpec.describe Notifications::WelcomeNotificationJob, type: :job do
+  describe "::perform_now" do
+    let(:user) { double }
+    let(:welcome_notification_service) { Notifications::WelcomeNotification::Send }
```
Comment (lightalloy) [inline spec/jobs/notifications/welcome_notification_job_spec.rb]: What do you think about using `double` as a  `welcome_notification_service` + pass it to the `perform_now`?
It's more explicit in my opinion.

File: spec/services/notifications/welcome_notification/send_spec.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  let(:dev_account) { create(:user) }
+  let(:welcome_broadcast) { create(:broadcast, :onboarding) }
+  let(:welcome_notification) { welcome_notification_service.call(user.id, welcome_broadcast) }
+
+  describe "::call" do
+    before do
+      allow(User).to receive(:dev_account).and_return(dev_account)
+    end
+
+    it "checks a newly created welcome notification", :aggregate_failures do
```
Comment (lightalloy) [inline spec/services/notifications/welcome_notification/send_spec.rb]: Not a strict one: here is only one test, so `let`s are not that useful. I think the test would be easier to understand if the service was called in the `it` block.

File: spec/jobs/notifications/welcome_notification_job_spec.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    it "calls the service" do
+      allow(user).to receive(:id).and_return(Faker::Number.number(4))
+      allow(welcome_notification_service).to receive(:call)
+      described_class.perform_now(user.id)
+      expect(welcome_notification_service).to have_received(:call).once
+    end
+  end
+
+  describe "::perform_later" do
+    ActiveJob::Base.queue_adapter = :test
```
[the remark this answers is not in the document]
Comment (cyrillefr) [reply spec/jobs/notifications/welcome_notification_job_spec.rb]: I removed queue_adapter setting and added shared_examples

File: spec/jobs/notifications/welcome_notification_job_spec.rb (hunk 1 above)
Comment (cyrillefr) [reply spec/jobs/notifications/welcome_notification_job_spec.rb]: I have changed that spec accordingly.

File: spec/services/notifications/welcome_notification/send_spec.rb (hunk 2 above)
Comment (cyrillefr) [reply spec/services/notifications/welcome_notification/send_spec.rb]: Yes, I did not notice that one :). I have removed the lets.

**Comment by benhalpern** [comment]
Thanks a lot!

**Pull request merged**